### DISCUSSION TITLE
getting vertex coordinates from lines

### DISCUSSION THREAD

**Discussion by firstkingofrome** [pyvista/pyvista#2677] [Q&A] [answered]
Hello, I have a polyData object. I would like to be able to iterate over it so that I get the coordinates (in order) of all of the vertices that connect the lines and am unsure of how to do this. The documentation says that polydata.lines is a numpy array corresponding to the lines and I am unsure of exactly what this means.

For example lets say I make the object:

```
points = np.array([[0, 0, 0], [1, 0, 0]])
poly = pyvista.lines_from_points(points)
poly.plot()
```
I get a line segment connecting the two points as expected.

However if I inspect 
`poly.lines`

I get an array containing array([2, 0, 1]).  How can I use this to map back to get the coordinates (0,0,0) and (1,0,0) which actually actually make up the line?

Thank You, I appreciate the clarification!

**Comment by darikg** [comment, marked as answer]
The structure is a little clearer after reshaping:

```python
poly = pyvista.lines_from_points(np.zeros((5, 3))
poly.lines.reshape(-1, 3)

[[2, 0, 1],
 [2, 1, 2],
 [2, 2, 3],
 [2, 3, 4]]
```

The `2`s indicate the number of points in each cell. The two numbers following each `2` are vertex indices. (The lines array uses the same structure as cell arrays, which support arbitrarily sized cells, but every line has has 2 indices, so it's a little redundant.) 

So I guess you want something like: (assuming all the lines are contiguous)

```python
line_segments = poly.lines.reshape(-1, 3)[:, 1:]  # [[a, b], [b, c], [c, d], ...]
vertex_path_idx = np.append(line_segments[:, 0], line_segments[-1, 1])  # [a, b, c, d, ...]
```
